### PR TITLE
Adding `UnbiasedLinear` (linear without bias).

### PR DESCRIPTION
**Pull request by Narsil**
- Propose to add a different layer. It's uncommon enough that I don't
  think polluting `Linear` with bias being `Option<Tensor>` should be
ok.
- But it does come in handy for the LanguageModel head within
  `transformers`. I don't have a lot of other useful examples tbh.

In theory we could use a regular Linear and just init the bias to zero,
but that's a waste of operations afterwards.

### CODE REVIEW

**Comment by Narsil** [comment]
@coreylowman

File: src/nn/unbiased_linear.rs [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    type Error = D::Err;
+
+    /// 1d forward using [matmul()] and [add()].
+    fn try_forward(&self, x: T) -> Result<Self::Output, D::Err> {
+        x.try_matmul(self.weight.retaped::<T::Tape>().try_permute()?)
+    }
+}
+
+#[derive(Clone, Debug)]
+struct Bias1D<'a, const M: usize, E: Dtype, D: DeviceStorage> {
```
Comment (coreylowman) [inline src/nn/unbiased_linear.rs:110]: You can remove this since this doesn't have bias 😀

File: src/nn/unbiased_linear.rs [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        }
+    }
+}
+
+impl<const I: usize, const O: usize, E: Dtype, D: Device<E>, T> Module<T>
+    for UnbiasedLinear<I, O, E, D>
+where
+    T: SplitTape + TryMatMul<Tensor<Rank2<I, O>, E, D, T::Tape>> + HasErr<Err = D::Err>,
+    T::Tape: Tape<D>,
+    for<'a> Bias1D<'a, O, E, D>: Module<T::Output, Output = T::Output, Error = D::Err>,
```
Comment (coreylowman) [inline src/nn/unbiased_linear.rs:98]: You can remove this bound

**Review by coreylowman — CHANGES_REQUESTED** [review]
Looks good, after removal of the Bias1D struct we can merge!

File: src/nn/unbiased_linear.rs (hunk 1 above)
Comment (Narsil) [reply src/nn/unbiased_linear.rs]: Done !

File: src/nn/unbiased_linear.rs (hunk 2 above)
Comment (Narsil) [reply src/nn/unbiased_linear.rs]: Done !

**Review by coreylowman — APPROVED** [review]
Nice!!

**Pull request merged**